core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -363,7 +363,7 @@ class ZooKeeperClient(connectString: String,
     stateChangeHandlers.values.foreach(callBeforeInitializingSession _)
 
     inWriteLock(initializationLock) {
-      if (!connectionState.isAlive) {
+      if (!connectionState.isConnected) {
```

Comment:
@leiboo : Thanks for the PR. Not sure if this change helps. I posted the results from a simple program on the jira and it seems that if a ZK session expires, the connectionState is CLOSED, instead of CONNECTING.